tree/ntuple/v7/src/RFieldBase.cxx

```diff
@@ -617,26 +575,39 @@ ROOT::Experimental::RFieldBase::Create(const std::string &fieldName, const std::
                auto field = Create(memberDesc.GetFieldName(), memberDesc.GetTypeName(), options, desc, id).Unwrap();
                memberFields.emplace_back(std::move(field));
             }
-            auto recordField = Internal::CreateEmulatedField(fieldName, std::move(memberFields), canonicalType);
+            auto recordField =
+               Internal::CreateEmulatedField(fieldName, std::move(memberFields), fieldDesc.GetTypeName());
+            recordField->fTypeAlias = fieldDesc.GetTypeAlias();
             return recordField;
          }
       }
    } catch (RException &e) {
       auto error = e.GetError();
       if (createContext.GetContinueOnError()) {
-         return std::unique_ptr<RFieldBase>(std::make_unique<RInvalidField>(fieldName, canonicalType, error.GetReport(),
+         return std::unique_ptr<RFieldBase>(std::make_unique<RInvalidField>(fieldName, resolvedType, error.GetReport(),
                                                                             RInvalidField::RCategory::kGeneric));
       } else {
          return error;
       }
    }
 
    if (result) {
-      if (typeAlias != canonicalType)
-         result->fTypeAlias = typeAlias;
+      const TClassEdit::EModType modType = static_cast<TClassEdit::EModType>(
+         TClassEdit::kDropStlDefault | TClassEdit::kDropComparator | TClassEdit::kDropHash);
+      std::string normOrigType{typeName};
+      TClassEdit::TSplitType splitname(normOrigType.c_str(), modType);
+      splitname.ShortType(normOrigType, modType);
+      // See TClassEdit::GetNormalizedName
+      if (normOrigType.length() > 2 && normOrigType[0] == ':' && normOrigType[1] == ':') {
+         normOrigType.erase(0, 2);
+      }
+      normOrigType = Internal::GetRenormalizedTypeName(normOrigType);
```

Comment:
Side note: both `TClassEdit::TSplitType` and `Internal::GetRenormalizedTypeName` are tokenizing the template name.

Reply:
Added a TODO to `TokenizeTypeList`